src/battle_script_commands.c

```diff
@@ -8433,69 +8439,48 @@ static bool32 HasAttackerFaintedTarget(void)
         return FALSE;
 }
 
-static void HandleTerrainMove(u16 move)
+static bool8 HandleTerrainMove(u16 fieldStatus)
 {
-    u32 statusFlag = 0;
+    u8 i, multistring = 0;
 
-    switch (gBattleMoves[move].effect)
+    switch (fieldStatus)
     {
-    case EFFECT_MISTY_TERRAIN:
-        statusFlag = STATUS_FIELD_MISTY_TERRAIN;
-        gBattleCommunication[MULTISTRING_CHOOSER] = 0;
-        break;
-    case EFFECT_GRASSY_TERRAIN:
-        statusFlag = STATUS_FIELD_GRASSY_TERRAIN;
-        gBattleCommunication[MULTISTRING_CHOOSER] = 1;
-        break;
-    case EFFECT_ELECTRIC_TERRAIN:
-        statusFlag = STATUS_FIELD_ELECTRIC_TERRAIN;
-        gBattleCommunication[MULTISTRING_CHOOSER] = 2;
-        break;
-    case EFFECT_PSYCHIC_TERRAIN:
-        statusFlag = STATUS_FIELD_PSYCHIC_TERRAIN;
-        gBattleCommunication[MULTISTRING_CHOOSER] = 3;
+    case STATUS_FIELD_PSYCHIC_TERRAIN:
+        multistring++;
+    case STATUS_FIELD_ELECTRIC_TERRAIN:
+        multistring++;
+    case STATUS_FIELD_GRASSY_TERRAIN:
+        multistring++;
+    case STATUS_FIELD_MISTY_TERRAIN:
+        if (gFieldStatuses & fieldStatus)
+            return FALSE;
+        gBattleCommunication[MULTISTRING_CHOOSER] = B_MSG_TERRAIN_MISTY + multistring;
```

Comment:
I don't like this use of fall through, I don't think it would be clear to new or intermediate people that they should add a new terrain at the start of these `case`s. (I got this wrong when writing my review and came back to correct it).

Reply:
The fall-through was already there; I just reversed the order. I added a comment to say "ADD NEW TERRAINS HERE" at the top so I think that's clear enough, no?

Reply:
That's not what the diff is showing? The code was better before. Although it should have used named the `B_MSG_TERRAIN_x` constants instead of `0`/`1`/`2`/`3`.

Reply:
Oops, my memory isn't so great at 2am (should have just _looked_). I'll just make each case pick the chooser